test/targets.cpp

```diff
@@ -50,4 +53,21 @@ TEST_CASE(targets)
     EXPECT(ts.size() >= 1);
 }
 
+TEST_CASE(concurrent_targets)
+{
+    std::vector<std::thread> threads;
+
+    for(auto i = 0u; i < 10000; i++)
+    {
+        auto thread_body = []() {
+            auto ref_target = migraphx::make_target("gpu");
+            migraphx::register_target(ref_target);
```

Comment:
This is a data race even with mutexes in both `make_target()` and `register_target()`.  make_target() accesses the `target_map()` via the destructor `~target_handler()` in register_target.hpp, and register_target() accesses it directly.  How can we ensure that access to both methods together is made atomic or protected by the same lock?